src/App.tsx

```diff
@@ -15,36 +17,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  SORT_BY_ALPHABET = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+}
+
 export const App: React.FC = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortedBtn, setSortedBtn] = useState('');
+  const prepareGoods = () => {
+    const newGoods = [...goodsFromServer];
+
+    if (sortedBtn) {
+      newGoods.sort((good1, good2) => {
+        switch (sortedBtn) {
+          case SortBy.SORT_BY_ALPHABET:
+            return good1.localeCompare(good2);
+          case SortBy.SORT_BY_LENGTH:
+            return good1.length - good2.length;
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (isReverse) {
+      newGoods.reverse();
+    }
+
+    return newGoods;
+  };
+
+  const finalGoods = prepareGoods();
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() => setSortedBtn(SortBy.SORT_BY_ALPHABET)}
+          className={classNames('button is-info', {
+            'is-light': sortedBtn !== SortBy.SORT_BY_ALPHABET,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={() => setSortedBtn(SortBy.SORT_BY_LENGTH)}
+          className={classNames('button is-success', {
+            'is-light': sortedBtn !== SortBy.SORT_BY_LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          onClick={() => setIsReverse(!isReverse)}
+          className={classNames('button is-warning', {
+            'is-light': !isReverse,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortedBtn || isReverse) && (
+          <button
+            onClick={() => {
+              setSortedBtn('');
+              setIsReverse(false);
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```